application/common/sync/Synchronizer.php

```diff
@@ -289,7 +408,35 @@ public function syncAll()
             }
         }
         
-        /** @todo Add a safety check here to avoid deactivating too many users. */
+        $numChangesAllowed = $this->getNumChangesAllowed($numActiveUsersInBroker);
+        
+        if (count($usersToAdd) > $numChangesAllowed) {
+            $this->abortSync(
+                'create',
+                count($usersToAdd),
+                $numChangesAllowed,
+                $numActiveUsersInBroker,
+                1501165932
+            );
+        }
+        
+        /**
+         * NOTE: If we begin intelligently comparing users found in both places
+         * and only updating those that need it, then we could limit updates to
+         * the safety cutoff as well. For now, we try to update all users found
+         * to exist in both places, so for a full sync that will almost always
+         * be above the safety cutoff.
+         */
+        
+        if (count($employeeIdsToDeactivate) > $numChangesAllowed) {
+            $this->abortSync(
+                'deactivate',
+                count($employeeIdsToDeactivate),
+                $numChangesAllowed,
+                $numActiveUsersInBroker,
+                1499971625
+            );
+        }
         
         $this->createUsers($usersToAdd);
         $this->activateAndUpdateUsers($usersToUpdateAndActivate);
```

Comment:
Unfortunately, every user that exists in both the ID Store and ID Broker will be considered an update (when doing a full sync), so I don't know how many it's actually going to change. And that's the number I would need to be able to know when to abort.